exporting/prometheus/remote_write/remote_write.c

```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "remote_write.h"
+
+static int as_collected;
+static int homogeneous;
+char context[PROMETHEUS_ELEMENT_MAX + 1];
+char chart[PROMETHEUS_ELEMENT_MAX + 1];
+char family[PROMETHEUS_ELEMENT_MAX + 1];
+char units[PROMETHEUS_ELEMENT_MAX + 1] = "";
+
+/**
+ * Send header to a server
+ *
+ * @param sock a communication socket.
+ * @param instance an instance data structure.
+ */
+void prometheus_remote_write_send_header(int *sock, struct instance *instance)
+{
+    int flags = 0;
+#ifdef MSG_NOSIGNAL
+    flags += MSG_NOSIGNAL;
+#endif
+
+    struct prometheus_remote_write_specific_config *connector_specific_config =
+        instance->config.connector_specific_config;
+
+    static BUFFER *header;
+    if (!header)
+        header = buffer_create(0);
+
+    buffer_sprintf(
+        header,
+        "POST %s HTTP/1.1\r\n"
+        "Host: %s\r\n"
+        "Accept: */*\r\n"
+        "Content-Length: %zu\r\n"
+        "Content-Type: application/x-www-form-urlencoded\r\n\r\n",
+        connector_specific_config->remote_write_path,
+        instance->engine->config.hostname,
+        buffer_strlen((BUFFER *)instance->buffer));
+
+    send(*sock, buffer_tostring(header), buffer_strlen(header), flags);
```

Comment:
You are not checking the return size of the send to make sure the bytes are written

Reply:
Fixed.